### PR TITLE
[WIP] password reset

### PR DESCRIPTION
**Pull request by jsachs**
#### What's this PR do?
Adds password reset functionality, including sending emails with a reset link.

#### Related JIRA tickets:
SER-8

#### How should this be manually tested?
For now, by running the test harness. You can submit API requests to go through the workflow, but make sure to set `NODE_ENV=test`.

#### Any background context you want to provide?
Currently, the test to actually update the password with the reset token is failing. It looks like two things are going on. (1) in the unit test, running `user.reload()` doesn't pull in the password change. (2) it looks like the password is getting changed, but the new `salt` is not getting saved, and I am not sure why.

### CODE REVIEW

**Comment by au2** [comment]
these are quick style issues for the tests.  did not run anything yet.  1 and 2 may help with async issues.

1) supertest-as-promised is deprecated (see the repo). switch to supertest; it supports promises out of box.
2) do not use 'done' in your tests.  just return promises (i.e 'return request. ...', 'return ').  mocha has been supporting promises for a while now.
3) mocha does not recommend arrow functions 'https://mochajs.org/#arrow-functions'.  rr has a couple of airbnb settings to help if you choose to do this.  it also makes tests more readable.
4) Also looks like linting is not done for tests based on airbnb settings.

File: server/controllers/auth.controller.js [hunk 1]
```diff
@@ -57,12 +62,53 @@ function updatePassword(req, res, next) {
...
-export default { login, logout, updatePassword };
+/**
+ * Returns 200 OK if password was reset successfully
+ * @param req
+ * @param res
+ * @param next
+ * @returns {*}
+ */
+function resetToken(req, res, next) {
+    const userLine = 'You have requested the reset of the password for your GFTrade account';
```
Comment (Perry5) [inline server/controllers/auth.controller.js]: Why does this say GFTrade account?

Comment (jsachs) [reply server/controllers/auth.controller.js]: Good catch, it was just a copypasta from me.

**Comment by jsachs** [comment]
@au2 I am working on removing `done` from as many tests as possible. However, it looks like your RecReg tests use arrow functions. What are the advantages/disadvantages? Also, the tests are definitely linted and beautified, using the `.estlint` in the repo.

**Comment by au2** [comment]
Main reason is this link https://mochajs.org/#arrow-functions; i would rather follow library recommendations although I do not think we are or will be using context.  RecReg at some point went through massive eslint auto fixes.  I tried to do away arrow functions after that as much as possible for mocha tests but still a lot left.  Personally (from RecReg experience) arrow functions makes it harder to read the tests but that is me.

You are right eslint is good.  My editor was showing errors (since I did not do npm install) so I thought it was not run.  I saw a console.log but but otherwise good.

**Comment by jsachs** [comment]
@au2 we have narrowed down the issue to the new `salt` not being saved when a password is updated. It looks like it also happens with the `password-update` endpoint. The tests pass, but I don't think the new salt is actually saved to the database. Password updates are done in a `beforeValidate` hook. Is this not the correct place for this?

**Comment by au2** [comment]
It should be good enough although it looks like we changed the hooks we use in recruitment registry.  i need to take a closer look maybe tonight; i do not see an issue looking at the code. btw you are using the old version of sequelize (3).  since this is a new service i suggest you use the new version (4).  if you do that you probably want recruitment registry version of this functionality. we do it better rr i think.

**Comment by au2** [comment]
1) Looks like there are issues with beforeValidate reintroduced recently https://github.com/sequelize/sequelize/issues/7903.  Not sure if related to issues you are seeing.
2) I do not think you are checking if update-password worked.  You need to login with the new password to make sure password update worked.
3) Same with 2 but for reset password.

**Comment by au2** [comment]
Also I am not sure how this will tie to recreg but recreg uses cookies instead of res.body for sending back jwt.  That was a conscience Sean/Mike decision which is more common practice nowadays I believe.  You might want to check.  I am also assuming that you will add authorization to post user at some point.

**Comment by jsachs** [comment]
@au2 good catches. We noticed problems with points (2) and (3) above, which I think are related to the salt not getting saved. I'm investigating if it has anything to do with the issue you linked.

I will make a separate issue for using cookies, I agree that's the right idea. We are going to auth protect more of the routes in a later sprint when the other devs have gotten used to integrating with the auth service.

**Pull request merged**